pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,382 @@
+
+        self.scale = size
+        self.setPath(self._path)
+        self.setLabel(label, labelOpts)
+
+    def setPos(self, pos):
+        if isinstance(pos, tuple):
+            newPos = pos
+        elif isinstance(pos, list):
+            newPos = tuple(pos)
+        elif isinstance(pos, (QtCore.QPointF, QtCore.QPoint)):
+            newPos = (pos.x(), pos.y())
+        else:
+            raise TypeError
+        if self._pos != newPos:
+            self._pos = newPos
+            super().setPos(Point(self._pos))
+            self.sigPositionChanged.emit(self)
+
+    def setBrush(self, *args, **kwargs):
+        """Set the brush that fills the symbol. Allowable arguments are any that
+        are valid for :func:`~pyqtgraph.mkBrush`.
+        """
+        self.brush = fn.mkBrush(*args, **kwargs)
+        if not self.mouseHovering:
+            self.currentBrush = self.brush
+            self.update()
+
+    def setHoverBrush(self, *args, **kwargs):
+        """Set the brush that fills the symbol when hovering over it. Allowable
+        arguments are any that are valid for :func:`~pyqtgraph.mkBrush`.
+        """
+        self.hoverBrush = fn.mkBrush(*args, **kwargs)
+        if self.mouseHovering:
+            self.currentBrush = self.hoverBrush
+            self.update()
+
+    def setPen(self, *args, **kwargs):
+        """Set the pen for drawing the symbol. Allowable arguments are any that
+        are valid for :func:`~pyqtgraph.mkPen`."""
+        self.pen = fn.mkPen(*args, **kwargs)
+        if not self.mouseHovering:
+            self.currentPen = self.pen
+            self.update()
+
+    def setHoverPen(self, *args, **kwargs):
+        """Set the pen for drawing the symbol when hovering over it. Allowable
+        arguments are any that are valid for
+        :func:`~pyqtgraph.mkPen`."""
+        self.hoverPen = fn.mkPen(*args, **kwargs)
+        if self.mouseHovering:
+            self.currentPen = self.hoverPen
+            self.update()
+
+    def setLabel(self, text=None, labelOpts=None):
+        if text is None and self.label is not None:
+            # remove the label if it's already added
+            if self.label.scene() is not None:
                 self.label.scene().removeItem(self.label)
-                self.label = None
+            self.label = None
         else:
+            if isinstance(text, bool):
+                # convert to default value or empty string
+                text = "x = {: .3n}\ny = {: .3n}" if text else ""
             if self.label is None:
-                self.label = TextItem()
-                self.label.setParentItem(self)
-            self.label.setText(label)
+                # add a label
+                labelOpts = {} if labelOpts is None else labelOpts
+                self.label = TargetLabel(self, text=text, **labelOpts)
+            else:
+                # update the label text
+                self.label.format = text
```

Comment:
A lot of those `kwargs` work because it's during the `TextItem` initializer, they all don't map nicely to being updated; for that you would nee do call the appropriate setters; I'm not seeing a good way that `.setLabel()` can try and pass those along in a way they can be effective.  Do you happen to have a way in mind?